### PR TITLE
Optimize ModelHelper and Use Class Wrapper instead of Copying

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

This PR optimizes `ModelHelper` by replacing calls that construct a static `Pattern` on each invocation to using a static `Pattern` instead and uses a wrapper-style pattern when translating a code generated class into the return public API class. The wrapper-style pattern becomes more important with upcoming code generation changes which use a deferred deserialization pattern which the current eager access and set pattern for translation removes any deferred deserialization benefit.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-blob-nio/README.md [hunk 1]
```diff
@@ -191,13 +191,6 @@ Write to a file. Only writing whole files is supported. Random IO is not support
 to guarantee that the data is available to be read.
 
 ```java readme-sample-writeToAFile
-try (OutputStream os = Files.newOutputStream(filePath)) {
-    os.write(0);
-}
-```
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob-nio/README.md:197]: nit: this looks like a bug in the codesnippet tooling where a trailing whitespace, possibly any character, in the closing code fence causes the code fence close to be ignored.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/accesshelpers/BlobPropertiesAccessHelper.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.implementation.accesshelpers;
+
+import com.azure.storage.blob.implementation.models.BlobPropertiesInternal;
+import com.azure.storage.blob.models.BlobProperties;
+
+public final class BlobPropertiesAccessHelper {
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/accesshelpers/BlobPropertiesAccessHelper.java:9]: javadoc

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/BlobPropertiesInternalGetProperties.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+import com.azure.storage.blob.models.RehydratePriority;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of {@link BlobPropertiesInternal} that represents the get properties {@link BlobProperties}.
+ */
+public final class BlobPropertiesInternalGetProperties implements BlobPropertiesInternal {
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/BlobPropertiesInternalGetProperties.java:26]: Why do we need this one and the constructor one?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ChunkedDownloadUtils.java [hunk 4]
```diff
@@ -40,7 +40,7 @@
             ? range.getCount() : parallelTransferOptions.getBlockSizeLong();
 
         return downloader.apply(new BlobRange(range.getOffset(), initialChunkSize), requestConditions)
-            .subscribeOn(Schedulers.elastic())
+            .subscribeOn(Schedulers.boundedElastic())
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ChunkedDownloadUtils.java:43]: Can you leave a comment about why we selected this scheduler so we aren't guessing about it in the future?

**Review by rickle-msft — APPROVED** [review]

**Pull request merged**